src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -433,6 +433,12 @@ private InsertValidationResponse insertRowsWithFallback(StreamingBuffer buffer)
     Fallback<Object> reopenChannelFallbackExecutorForInsertRows =
         Fallback.builder(() -> insertRowsFallbackSupplier(buffer))
             .handle(SFException.class)
+            .onFailedAttempt(
+                event ->
+                    LOGGER.warn(
+                        String.format(
+                            "Failed Attempt to invoke the insertRows API for buffer:%s", buffer),
```

Comment:
InsertRows shouldn't fail other than invalid or closed channel, what's the reason that you saw?

Reply:
I did see 429 from one of the APIs. (I think it was blob upload)

Reply:
what do you mean by 429?

Reply:
Sorry I meant 429 response code: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429

Reply:
hmmm, that's weird. We're not calling any REST APIs in the InsertRows API, i assume we're talking about `SnowflakeStreamingIngestChannelInternal.insertRows`?